pkg/util/validation/validation.go

```diff
@@ -242,24 +242,33 @@ var httpHeaderNameRegexp = regexp.MustCompile("^" + httpHeaderNameFmt + "$")
 // definition of a valid header field name (a stricter subset than RFC7230).
 func IsHTTPHeaderName(value string) []string {
 	if !httpHeaderNameRegexp.MatchString(value) {
-		return []string{RegexError(httpHeaderNameFmt, "X-Header-Name")}
+		return []string{RegexError(httpHeaderNameFmt, true, "X-Header-Name")}
 	}
 	return nil
 }
 
-const configMapKeyFmt = "\\.?" + dns1123SubdomainFmt
+const configMapKeyFmt = `\A[.\-_a-zA-Z0-9]+[^.]?\z`
```

Comment:
if you put the - as the first char of a character class you don' t need to escape it